hoodie-common/src/main/java/com/uber/hoodie/common/model/HoodieAvroPayload.java

```diff
@@ -29,10 +30,16 @@
  */
 public class HoodieAvroPayload implements HoodieRecordPayload<HoodieAvroPayload> {
 
-  private final Optional<GenericRecord> record;
+  // Store the GenericRecord converted to bytes - 1) Doesn't store schema hence memory efficient 2) Makes the payload
+  // serializable
+  private final byte [] recordBytes;
 
   public HoodieAvroPayload(Optional<GenericRecord> record) {
-    this.record = record;
+    try {
+      this.recordBytes = HoodieAvroUtils.avroToBytes(record.get());
```

Comment:
So to do that for payloads, it's tricky since we would have to have the binaryEncoder as a member variable along with the outputStream. That would mean we would have to introduce another "non-thread safe" method. Also, the BinaryEncoder itself is not thread safe : https://github.com/apache/avro/blob/release-1.7.7/lang/java/avro/src/main/java/org/apache/avro/io/EncoderFactory.java#L153.
We could introduce this method and make use of it in the compaction step since we are single threaded there but for HoodieAvroPayload use-cases may be not given people could init the Payload in a multi-threaded way.
WDYT ?

Reply:
We could have a static encoder cache per schema right? I think we should address this if the performance hit is large ..